providers/base/units/firmware/jobs.pxu

```diff
@@ -105,3 +105,15 @@ plugin: attachment
 depends: firmware/fwts_dump
 command:
   [ -f "$PLAINBOX_SESSION_SHARE/acpidump.log" ] && gzip -c "$PLAINBOX_SESSION_SHARE/acpidump.log"
+
+id: firmware/fwupdmgr_get_devices
+plugin: attachment
+category_id: com.canonical.plainbox::firmware
+_summary: Collect the device firmware update information
+requires:
+    executable.name == "fwupdmgr"
+command:
+    snap_var="$SNAP"
+    unset SNAP
+    fwupdmgr get-devices --json
+    export SNAP="$snap_var"
```

Comment:
I'm not sure the specific purpose of the test, but if it's to check end to end updates or supportability you probably should also be running `fwupdmgr refresh` as well to ensure the supported flag is accurately set.  Just make sure you catch the return code of 2 (nothing to do) in case it's up to date already.  something like:

```
fwupdmgr refresh --json || $? == 2
```

If it's used in pre-production there should be a step to add an embargo remote too if it doesn't exist.

Reply:
Mario,

the testing purpose of this job is to capture the all devices and current firmware version detected by fwupd. And I think the main reason to log these information, is we would like to compare the firmware version if any issues happens on a specific system.